datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -121,13 +148,18 @@ protected Duration _fromString(JsonParser parser, DeserializationContext ctxt,
             }
             return null;
         }
+
         // 30-Sep-2020: Should allow use of "Timestamp as String" for
         //     some textual formats
         if (ctxt.isEnabled(StreamReadCapability.UNTYPED_SCALARS)
                 && _isValidTimestampString(value)) {
             return _fromTimestamp(ctxt, NumberInput.parseLong(value));
         }
 
+        if (pattern != null) {
+            return pattern.parse(NumberInput.parseLong(value));
```

Comment:
Exception handling tested?

Reply:
I generalized the handling of `DateTimeException` in the base class and tested the code.